legend-engine-executionPlan-execution-store-relational-connection-interfaces/src/main/java/org/finos/legend/engine/plan/execution/stores/relational/connection/ds/DataSourceSpecification.java

```diff
@@ -0,0 +1,315 @@
+    {
+        try (Scope scope = GlobalTracer.get().buildSpan("Create Pool").startActive(true)) {
+            String currentUser = SubjectTools.getCurrentUsername() != null ? SubjectTools.getCurrentUsername() : this.authenticationStrategy.getAlternativePrincipal(profiles);
+            String poolName = String.format("DBPool_%s_%s", this.connectionKey.shortId(), currentUser);
+
+            Properties properties = new Properties();
+            properties.putAll(this.buildJdbcDatasourceProperties());
+
+            // TODO : epsstan : Why is the profile registered here
+            properties.put(AuthenticationStrategy.AUTHENTICATION_STRATEGY_KEY, this.authenticationStrategy.getKey().shortId());
+            properties.put(AuthenticationStrategy.AUTHENTICATION_STRATEGY_PROFILE_BY_POOL, poolName);
+            // The profiles are associated with the Pool as the Pool can create connections by itself in its own threads.
+            AuthenticationStrategy.registerProfilesByPool(poolName, profiles);
+
+            HikariConfig jdbcConfig = new HikariConfig();
+            jdbcConfig.setDriverClassName(databaseManager.getDriver());
+            jdbcConfig.setPoolName(poolName);
+            jdbcConfig.setMaximumPoolSize(HIKARICP_MAX_POOL_SIZE);
+            jdbcConfig.setMinimumIdle(HIKARICP_MIN_IDLE);
+            jdbcConfig.setJdbcUrl(this.buildJdbcURL());
+            jdbcConfig.setConnectionTimeout(authenticationStrategy.getConnectionTimeout());
+            jdbcConfig.addDataSourceProperty("cachePrepStmts", false);
+            jdbcConfig.addDataSourceProperty("prepStmtCacheSize", 0);
+            jdbcConfig.addDataSourceProperty("prepStmtCacheSqlLimit", 0);
+            jdbcConfig.addDataSourceProperty("useServerPrepStmts", false);
+            jdbcConfig.addDataSourceProperty("privateProperty", "MyProperty");
+
+            // jdbcConfig.setHealthCheckRegistry(new HealthCheckRegistry());
+            if (this.databaseManager.publishMetrics()) {
+                jdbcConfig.setMetricRegistry(METRIC_REGISTRY);
+            }
+
+            // Properties management --------------
+            MapAdapter.adapt(properties).keyValuesView().forEach((Procedure<Pair>) p -> jdbcConfig.addDataSourceProperty(p.getOne().toString(), p.getTwo()));
+            //-------------------------------------
+
+            HikariDataSource ds = new HikariDataSource(jdbcConfig);
+            scope.span().setTag("Pool", ds.getPoolName());
+            LOGGER.info("Create new Connection Pool  {}", ds);
+            return ds;
+        }
+    }
+
+    public final Properties buildJdbcDatasourceProperties()
+    {
+        Properties properties = this.properties;
+        Properties extraProperties = this.buildExtraJdbcDatasourceProperties();
+        if (extraProperties != null)
+        {
+            this.properties.putAll(extraProperties);
+        }
+        return properties;
+    }
+
+    protected String buildJdbcURL()
```

Comment:
For scenarios where we have multiple drivers for a db type, this might be a bit messy.
Since this is really specifc to driver & dbType maybe we can move this to manager -> driver